---
title: "Assignment2"
author: "Ashenafi Beyi"
date: "3/19/2021"
output: html_document
---
Installing


Loading

```[r]
library("readr")
```
Import genotype data set
install readr

install.packages("readr")


```{r}
library(readr)
fang_genotypes <- read.table("C:/Data analysis/Bioinfo_spring2021/fang_et_al_genotypes.txt", header = T)
dim(fang_genotypes) 
```
2782 rows/observations and 986 columns/variables

Import snp_position data set
```{r}
library(readr)
snp_position <- read_tsv("C:/Data analysis/Bioinfo_spring2021/snp_position.txt")
class(snp_position)
str(snp_position) 
dim(snp_position) # 983 rows/observations and  15 columns/variables
head(snp_position) 
```


Data processing


install.packages("dplyr")


```{r}
library(dplyr)
snp_position.selected <- select(snp_position, c("SNP_ID", "Chromosome", "Position"))
str(snp_position.selected) # 983 rows and 3 columns
glimpse(snp_position.selected) # similar function with str
```

Number of rows and columns
```{r}
nrow(fang_genotypes) # 2782
ncol(fang_genotypes) # 986
NROW(na.omit(fang_genotypes)) # 2782
NCOL(na.omit(fang_genotypes)) # 986
```

Genotypes in the column Group, summary and table functions can be used alike.
```{r}
levels(as.factor(fang_genotypes$Group)) # 15 groups: "TRIPS" "ZDIPL" "ZLUXR" "ZMHUE" "ZMMIL" "ZMMLR" 
                                        # "ZMMMR" "ZMPBA" "ZMPIL" "ZMPJA" "ZMXCH" "ZMXCP" "ZMXIL" 
                                        # "ZMXNO" "ZMXNT" "ZPERR"
summary(as.factor(fang_genotypes$Group))
table(as.factor(fang_genotypes$Group))
```

Number of chromosomes in snp_position data
```{r}
summary(as.factor(snp_position$Chromosome))
```

Subset data 3 maize genotypes > transpose > merge with snp data
```{r}
maize_genotypes <- filter(fang_genotypes, Group == 'ZMMIL' | Group == 'ZMMLR' | Group == 'ZMMMR')
dim(maize_genotypes) # 1573 rows/observations  986 columns/variables
levels(as.factor(maize_genotypes$Group)) # "ZMMIL" "ZMMLR" "ZMMMR"
summary(as.factor(maize_genotypes$Group)) # ZMMIL = 290, ZMMLR = 1256, ZMMMR = 27 
```

      #Transpose

resource: reading: https://tibble.tidyverse.org/reference/rownames.html

```{r}
library(tidyverse)
maize_genotypes <- column_to_rownames(maize_genotypes, var = "Sample_ID")
```

```{r}
maize_genotypes.tr <- t(maize_genotypes)%>%as.data.frame()%>%rownames_to_column(., var = "SNP_ID")
maize_genotypes.tr <- maize_genotypes.tr[3:nrow(maize_genotypes.tr),]

maize_snp <- merge(snp_position.selected, maize_genotypes.tr, by="SNP_ID")
maize_snp <- select(maize_snp, SNP_ID, Chromosome, Position, everything())
dim(maize_snp) # 983 528
table(as.factor(maize_snp$Chromosome))
```
Check for missed chromosome 
```{r}
sum(maize_snp$Chromosome == "") # 0
```

Subset data 3 teosinte genotypes > transpose > merge with snp data
```{r}
teosinte_genotypes <- filter(fang_genotypes, Group == 'ZMPBA' | Group == 'ZMPIL' | Group == 'ZMPJA')
dim(teosinte_genotypes) # 975 rows/observations and 986 columns/variables
levels(as.factor(teosinte_genotypes$Group)) # "ZMPBA" "ZMPIL" "ZMPJA"
summary(as.factor(teosinte_genotypes$Group)) # ZMPBA = 900, ZMPIL = 41, ZMPJA = 34
```

Transpose

```{r}
teosinte_genotypes <- column_to_rownames(teosinte_genotypes, var = "Sample_ID")
```

```{r}
teosinte_genotypes.tr <- t(teosinte_genotypes)%>%as.data.frame()%>%rownames_to_column(., var = "SNP_ID")
teosinte_genotypes.tr <- teosinte_genotypes.tr[3:nrow(teosinte_genotypes.tr),]

teosinte_snp <- merge(snp_position.selected, teosinte_genotypes.tr, by="SNP_ID")

teosinte_snp <- select(teosinte_snp, SNP_ID, Chromosome, Position, everything())

dim(teosinte_snp) # 983 528
```
The merged dataset has 983 rows/observations and 528 columns/variables.

check missed values in chromosome
```
Check for missed values
```{r}
sum(teosinte_snp$Chromosome == "") 
```


 In progress
Subset data by chromosome, order in increasing position

maize_chromosome1 <- subset(maize_snp, Chromosome==1)%>%arrange(Position)
teosinte_chromosome1 <- subset(teosinte_snp, Chromosome==1)%>%arrange(Position)
teosinte_chromosome2 <- subset(teosinte_snp, Chromosome==2)%>%arrange(teosinte_chromosome2$Position)


Subset each chromosome, order in position decreasing order, replace missed values

maize_chromosome.dec1 <- subset(maize_snp, Chromosome==1)%>%arrange(desc(Position))
teosinte_chromosome.dec1 <- subset(teosinte_snp, Chromosome==1)%>%arrange(desc(Position))

